internal/worker/rundynamic.go

```diff
@@ -129,6 +131,20 @@ func RunDynamicAnalysis(ctx context.Context, pkg *pkgmanager.Pkg, sbOpts []sandb
 	return result, nil
 }
 
+// getStraceLogger returns an slog.Logger instance for the strace parsing functions.
+// If featureflags.DisableStraceDebugLogging is enabled, the logger will discard any
+// log messages with level below slog.LevelInfo (i.e. LevelDebug). Otherwise, the
+// default logging method is used.
+func getStraceLogger(ctx context.Context) *slog.Logger {
+	logger := slog.Default()
+	if featureflags.DisableStraceDebugLogging.Enabled() {
+		infoOnly := &slog.HandlerOptions{Level: slog.LevelInfo}
+		logger = slog.New(slog.NewTextHandler(os.Stderr, infoOnly))
```

Comment:
I think it is a problem using a new handler here as it will be entirely different to the zap-based one used everywhere else - leading to all log lines using this logger being different when strace logging is disabled.

Unfortunately there isn't a simple way to filter out logs on a per-use basis.

One approach could be to have a pass-thru handler than applies the log level filtering, and a new logger can be created on the current logger's handler:

e.g.
```go
logger = slog.New(log.WrapHandlerAtLevel(logger.Handler(), slog.LevelInfo))
```

Reply:
Ok I think I'll just disable the strace logging entirely for now. The only difference is the two log.Warn() calls which I've never seen printed. I also don't think this feature should be enabled generally, it's more for local analysis usage. So I don't think it's a big deal to disable all strace logging via this feature.

Once we replace zap entirely with native slog, we might be able to implement the log level-based filtering more easily?

Reply:
Or, if we don't want to lose the warning logs, I can redirect everything to another file where the different format doesn't matter.

Reply:
I don't mind losing the logs in the `worker` binary. They are super noisy and not worth keeping.

We can `slog.WarnContext` to keep the warning logs.

Reply:
I've reworked this to remove the debug logging calls in the worker, and allow optionally saving them to a separate file in the analysis image. See updated original PR description above